src/ctia/http/routes/bundle.clj

```diff
@@ -113,17 +113,29 @@
       (transient-id? id) (assoc :original_id id)
       external_id (assoc :external_id external_id))))
 
+(def find-by-external-ids-limit 1000)
+
 (defn find-by-external-ids
   [import-data entity-type auth-identity]
   (let [external-ids (remove nil? (map :external_id import-data))]
     (log/debugf "Searching %s matching these external_ids %s"
                 entity-type
                 (pr-str external-ids))
     (if (seq external-ids)
-      (:data (read-store entity-type (list-fn entity-type)
-                       {:external_ids external-ids}
-                       (auth/ident->map auth-identity)
-                       {:limit (count external-ids)}))
+      (debug (format "Results for %s:" (pr-str external-ids))
+             (loop [paging {:offset 0
+                            :limit find-by-external-ids-limit}
+                    entities []]
+               (let [{results :data
+                      {next-page :next} :paging}
+                     (read-store entity-type (list-fn entity-type)
+                                 {:external_ids external-ids}
+                                 (auth/ident->map auth-identity)
+                                 paging)
+                     acc-entities (into entities results)]
+                 (if next-page
+                   (recur next-page acc-entities)
+                   acc-entities))))
```

Comment:
As loop are not easy to read, would you mind to write an helper?

Something similar to:

```clojure
(defn all-pages [fn]
   (loop [paging {:offset 0 :limit find-by-external-ids-limt}
              entities []]
              (let [{results :data {next-page :next} :paging} (fn paging) acc-entities (into entities results]
                  (if next-page
                         (recur next-page acc-entities)
                         acc-entities)))
```